Fixed Constants should be on the right side of comparison

LaithAlebrahim opened the pull request:
@yegor256 , Please can you review this [issue#716](https://github.com/yegor256/qulice/issues/716) and let me know if this what is required from the task.
I searched for all null comparison in the code and this what I found where null should be on right side of the comparsion.
Please let me know if you have any concerns .Tx

LaithAlebrahim committed:
Fixed Constants should be on the right side of comparison

LaithAlebrahim commented:
@yegor256 , Could you please review this and tell me if this is what meant in task.Thanks

yegor256 commented:
@pnatashap can you please check this one?

LaithAlebrahim committed:
Undo my last commit and revert it

LaithAlebrahim committed:
Fixed Constants should be on the right side of comparison

LaithAlebrahim commented:
Could you please now review it and check if everything is fine .And please let me know if you have any concerns.
@yegor256 @pnatashap 
This the link of the exact changes [LastCommit](https://github.com/yegor256/qulice/pull/1237/commits/7f20258d58c92c77f12136136835e0b8ec146cb8)

pnatashap commented on qulice-pmd/src/main/resources/com/qulice/pmd/ruleset.xml:
```diff
@@ -256,4 +256,26 @@ OF THE POSSIBILITY OF SUCH DAMAGE.
   </rule>
+  <rule name="ConstantsOnRightSideOfComparison" language="java" class="net.sourceforge.pmd.lang.rule.XPathRule" message="Constants should be on the right side of comparisons.">
+    <description>
+      Enforces the code style guideline that constants (null, string literals, or numbers) should appear on the right side of comparison operators to reduce the risk of accidental assignment and improve readability.
```
- It is better to have this rule only about numbers and null values (because for string values there is another rule, that will write that you need to use equals method instead of ==) (So if you have code `` "some" == some``, you will get the first warning to revert it, and the second - use equals and it can be confusing)
- The same rule should be about != operation, as they are similar, so ``null != some`` should produce the same error

pnatashap reviewed (changes requested):
And also test is needed to check the rule, like here  https://github.com/yegor256/qulice/blob/master/qulice-pmd/src/test/java/com/qulice/pmd/UseStringIsEmptyRuleTest.java

LaithAlebrahim committed:
Modify the rule to handle just numbers and null values

LaithAlebrahim commented:
@yegor256 @pnatashap  I modified the rule to handle the specific requirments just (numeral and null with '!=' and '==' equality)
(https://github.com/yegor256/qulice/pull/1237/commits/d102ee07a331a9f4a52731bccd35f379718966fd)
And added Todo for the testing I'll work on it soon.
Please review it and let me know if you have any concerns. Thanks

yegor256 commented:
@LaithAlebrahim would be great to add a test for this new feature, otherwise how do we know that it works?

pnatashap commented:
@LaithAlebrahim I have checked this rule and do not see any violations, so looks like something does not work

yegor256 commented:
We are closing this pull request, because it has merge conflicts with `master` and has been open for more than a month. Please, forgive us for this: it says nothing about the quality of your work, it's just housekeeping.

If the change still matters to you, please submit a new pull request. It may be very similar to this one, but it has to take into account everything that landed in `master` since then.

Thank you for contributing, and sorry again for the inconvenience!
